### DISCUSSION TITLE
max-query-depth: 10 configuration causing errors

### DISCUSSION THREAD

**Discussion by Jojoooo1** [graphql-java-kickstart/graphql-spring-boot#609] [Q&A]
**Describe the bug**
Set graphql.servlet.max-query-depth: 10

**To Reproduce**
1. Go to application.yaml set graphql.servlet.max-query-depth: 10
2. Go to /playground
3. Open developer console: "ServerParseError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data"

**Expected behavior**
Playground should be working

**Desktop (please complete the following information):**
- Browser: Firefox
- Version: 87.0 (64-bit)

**Comment by oliemansm** [comment]
I just tried this in an example project using the latest versions and I'm not getting those JSON parse errors. Instead I'm seeing in the console (expected) errors about max query depth being reached:

```
middleware.js:98 Uncaught Error: {
  "errors": [
    {
      "message": "maximum query depth exceeded 13 > 10",
      "extensions": {
        "classification": "ExecutionAborted"
      }
    }
  ],
  "data": null
}
    at Object.next (middleware.js:98)
    at b (middleware.js:98)
    at E (middleware.js:98)
    at e.value (middleware.js:98)
    at middleware.js:124
```

So from my point of view this is working as expected. Sounds like there's a specific problem with your local configuration instead. I'll convert this into a Discussion.

**Reply by Thinkenterprise** [reply]
@Jojoooo1 @oliemansm 
The problem occurs in class ``MaxQueryDepthInstrumentation``. The introspection query from Playground has a **query depth** of 13. This comes from the playground schema polling!! I try to disable the schema polling of playground over the application properties but it doesn`t work.

```
playground:
    enabled: true
    settings:
      schema:
        polling:
          enable: false`
```
A workaround could be to overwrite the class ``MaxQueryDepthInstrumentation``, adapt it and create it yourself using the Spring Boot configuration? Or there is a change in the kickstarter project that specializes the introspection case in the implemention of the method ``beginValidation``.

**Reply by oliemansm** [reply]
@Thinkenterprise I checked the `graphql-java` project issues and Discussions yesterday to see if there was anybody reporting issues using their `MaxQueryDepthInstrumentation` in combination with the introspection query. Because I can imagine that you'd want to exclude that query for this instrumentation. Couldn't find any open issues. Could be worth to start a discussion in their project to see if that's something they'd considered or are open to supporting (through configuration).

**Reply by Thinkenterprise** [reply]
@oliemansm I think that I found a discussion about that problem in graphql java https://github.com/graphql-java/graphql-java/issues/1055. There is probably no simple solution to solving the problem for everyone.

**Reply by Thinkenterprise** [reply]
@oliemansm I looked around again at Github Playground. There is also the problem that the introspection disable does not work. https://github.com/graphql/graphql-playground/issues/1250. Disableing could be a quick workaround!?

**Reply by oliemansm** [reply]
@Thinkenterprise You mean disabling the introspection query on GraphQL servlet level? Worth a try, although I wouldn't be surprised if Playground would stay in the same error loop it stays in now. 

The Docs section of Playground wouldn't work obviously without introspection, so that seems to be the main problem when using this instrumentation when you also want to use an IDE, because they all can't finish that introspection query. Do you use that instrumentation typically during development or actually more on production or something else?

**Reply by Thinkenterprise** [reply]
@oliemansm you are rigth. I could disable the DDOS Feature Query Depth during developing and enable it in production. But it should be possible to use a IDE  like Playground, GraphiQL to test an GraphQL API in production, right? If we can't disable the introspection query we could define a configuration for the depth of the introspection query. This does not have to be 13 !! But this is in the responsability of Playground
